src/readwrite.rs

```diff
@@ -0,0 +1,91 @@
+// Copyright 2018-2019 Mozilla
+//
+// Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+// this file except in compliance with the License. You may obtain a copy of the
+// License at http://www.apache.org/licenses/LICENSE-2.0
+// Unless required by applicable law or agreed to in writing, software distributed
+// under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+// CONDITIONS OF ANY KIND, either express or implied. See the License for the
+// specific language governing permissions and limitations under the License.
+
+use lmdb::{
+    Database,
+    RoCursor,
+    RoTransaction,
+    RwTransaction,
+    Transaction,
+    WriteFlags,
+};
+
+use crate::error::StoreError;
+use crate::read_transform;
+use crate::value::Value;
+
+pub struct Reader<'env>(pub RoTransaction<'env>);
+pub struct Writer<'env>(pub RwTransaction<'env>);
+
+pub trait Read {
+    fn get<K: AsRef<[u8]>>(&self, db: Database, k: &K) -> Result<Option<Value>, StoreError>;
+    fn open_ro_cursor(&self, db: Database) -> Result<RoCursor, StoreError>;
```

Comment:
Shall we make this function as `crate(pub)`? Looks like `RoCursor` is only internally used by the rkv iterators. If this makes sense, let's remove the `pub use lmdb::RoCursor` in the lib.rs.

Reply:
I haven't found a way to do this, as it seems like the visibility of the function depends on the visibility of the trait, and it isn't possible to hide the trait, since public methods like `SingleStore.get()` have parameters that are bound to the trait. Do you know of any way around that?

Reply:
Ugh, right, it's a trait. Perhaps split this trait into two, and make the cursor one visible at the crate level?

My concern is that exposing `RoCursor` to the consumer is not very useful in rkv, and it might lead to lmdb:errors upon misuses.

Reply:
I think we can polish this in a followup patch once we come up with the better solution.

Reply:
I don't think we can hide a trait that only declares an `open_ro_cursor()` function either, as we would still need to specify that trait in the public interfaces of `SingleStore.iter_start()`, `SingleStore.iter_from()`, and `MultiStore.get()` in order to call `open_ro_cursor()` on the Reader/Writer given to those functions.

Reply:
Yea, looks like hiding traits alone is just a deadend, perhpas apply the good old delegation trick to those iter_* functions? Need to stop myself from concerning this prematurely :)

Reply:
Hehe, I filed #117 so we don't forget about it! 😉